src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1269,7 +1266,28 @@ static bool addTriggeredSubscriptions_noCache
         std::string geometry = expr.hasField(CSUB_EXPR_GEOM)   ? getStringFieldF(expr, CSUB_EXPR_GEOM)   : "";
         std::string coords   = expr.hasField(CSUB_EXPR_COORDS) ? getStringFieldF(expr, CSUB_EXPR_COORDS) : "";
 
-        trigs->fillExpression(q, georel, geometry, coords);
+        trigs->fillExpression(georel, geometry, coords);
+
+        // Parsing q
+        if (q != "")
+        {
+          StringFilter* stringFilterP = new StringFilter();
+
+          if (stringFilterP->parse(q.c_str(), &err) == false)
+          {
+            delete stringFilterP;
+          
+            LM_E(("Runtime Error (%s)", err.c_str()));
+            TIME_STAT_MONGO_READ_WAIT_STOP();
```

Comment:
The issue is not about measuing times in failed request... is that the "end mark" for that time measure has been already done. In other words, you were calling twice in a row TIME_STAT_MONGO_READ_WAIT_STOP().

However, I may wrong. Please review the whole function to ensure that now is working properly for the different ways the function may `return`.